lib/AST/RequirementMachine/Rule.h

```diff
@@ -97,7 +97,6 @@ class Rule final {
   }
 
   void setRequirementID(Optional<unsigned> requirementID) {
```

Comment:
Is this ever called with None? Maybe the argument should be non-optional.

Reply:
Yes, from `RewriteSystem::addExplicitRule`. I believe it's because not all rewrite rules added via `RuleBuilder` are created from structural requirements